src/NUnitFramework/framework/Attributes/RepeatAttribute.cs

```diff
@@ -94,18 +94,33 @@ public override TestResult Execute(TestExecutionContext context)
 
                 while (count-- > 0)
                 {
-                    context.CurrentResult = innerCommand.Execute(context);
+                    try
+                    {
+                        context.CurrentResult = innerCommand.Execute(context);
+                    }
+                    // Commands are supposed to catch exceptions, but some don't
+                    // and we want to look at restructuring the API in the future.
+                    catch (Exception ex)
+                    {
+                        if (context.CurrentResult is null)
```

Comment:
CurrentResult is declared as non-nullable.
But TestExecutionContext doesn't explicitly set this to non-null.
Also see the comment on l.52 workitem.cs.  So, perhaps just accept this.......

Reply:
TestExecutionContext its properties is fake nullable. With a TODO:
```csharp
// TODO: Fix design where properties are not set at unknown times.

#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
```

It is literally copied from `RetryCommand.Execute`.
Looking at the code above, it assigns the result of the command to `context.CurrentResult`.
If the command fails, it is not assigned, unless it is assigned somewhere else previously which we cannot guarantee.